js/src/util/index.js

```diff
@@ -135,20 +135,26 @@ const typeCheckConfig = (componentName, config, configTypes) => {
 }
 
 const isVisible = element => {
-  if (!element) {
+  const eligibleElement = el => el && el.style
+
+  if (!eligibleElement(element)) {
+    return false
+  }
+
+  if (element.tagName !== 'AREA' && element.getClientRects().length === 0) {
     return false
   }
 
-  if (element.style && element.parentNode && element.parentNode.style) {
-    const elementStyle = getComputedStyle(element)
-    const parentNodeStyle = getComputedStyle(element.parentNode)
+  while (eligibleElement(element)) {
+    const { display, visibility } = getComputedStyle(element)
+    if (display === 'none' || visibility === 'hidden') {
```

Comment:
In theory this could return false negatives for `visibility` because an element can be visible by using `visibility: visible` while the parent has `visibility: hidden`.